src/util/util.js

```diff
@@ -124,6 +124,29 @@ function extractURLsFromTemplates(URLTemplates) {
   return URLTemplates;
 }
 
+/**
+ * Filter URLTemplates elements to keep only valid and safe URL templates.
+ *   To be valid, urls should:
+ *   - have the same protocol as the client
+ *   or
+ *   - be protocol-relative urls
+ *
+ * @param {String|String[]|Object[]} URLTemplates - A string|string[]|object[] of url templates.
```

Comment:
```suggestion
 * @param {Array} URLTemplates - An Array of url templates.
```
It should be only this no? As we are expecting an array all the time, and also this is the input value from `trackURLs`

Reply:
Good one. I needed to check after the urls that are not in an array such as survey. As far as I know, this is the only case, and it's not called using trackURLs. Update inc.